### ISSUE TITLE
bug: freeze and very very slow

### ISSUE THREAD

**Issue by Dev-AminMohebbi** [sourcegraph/cody#5780]
We have written the needed data into your clipboard because it was too large to send. Please paste.

**Comment by almeidapaulooliveira** [comment]
Same here, VS Code is **super slow** after updated to `1.36.0`. 
I uninstalled and confirmed that the extension was slowing down VS Code. I rolled back to version `1.34.3` to continue using it.

**Comment by sqs** [comment]
Sorry about this issue. Can you please paste the contents of the `Cody by Sourcegraph` VS Code output channel so we can see what might be causing it?

**Comment by PriNova** [comment]
Another user reported slowness if autocomplete is switched on and faster if switched off.
https://discord.com/channels/969688426372825169/969688427232641136/1291373158406033503
> when I have autocomplete enabled, I get a lot of stuttering when navigating in files, like typing, highlighting stuff, moving the cursor up and down.
when I turn off autocomplete I get none of that, I think it's new? I don't think I had this issue yesterday.

**Comment by arafatkatze** [comment]
Hi everyone,
To help us identify and resolve performance issues with Cody, we’d appreciate it if you could follow these steps to create a performance profile and share the generate file with us through a GitHub issue.

Steps to Create and Share a Performance Profile:

1. Enable the Extension Host Profiler in VS Code:
	•	Open VS Code.
	•	Press Ctrl+Shift+P (Windows/Linux) or Cmd+Shift+P (macOS) to open the Command Palette.
	•	Type Developer: Show Running Extensions and select it.
	•	A list of running extensions will appear. Next to Cody (or the extension you suspect is causing slowness), click rick click on it and click on _start extension host profile_.
2. Reproduce the Issue:
	•	While the profiler is running, perform the actions that typically cause the performance slowdown (e.g., chatting, using Cody, loading files).
	•	Once the issue has been reproduced, stop the profiler by returning to the list of running extensions and right clicking and doing _Stop Extension Host Profile_.
	3.	Save the Profile:
	•	Right click on Cody and click "save extension host profile " This will generate a .cpuprofile file. Save this file to a location on your computer.
	4.	Attach the Profile to a GitHub Issue:
	•	Attach the .cpuprofile file to this issue by uploading it along with a brief description of the issue you’re experiencing, including any specific steps that caused the slowdown.



I know this is a lot of steps but we have found this to be the case only with certain distributions and haven't been able to replicate it so we wanted to do this as its the best route to figure out the causality for your case. 

🙇

**Comment by justinechang39** [comment]
[CPU-20241003T122305.648Z.cpuprofile.txt](https://github.com/user-attachments/files/17244036/CPU-20241003T122305.648Z.cpuprofile.txt)

@arafatkatze 

<Steps taken when generating the profile>

Before the profile was started, autocomplete was switched off
Profiling started
autocomplete turned on
used arrow keys to move cursor up and down within a file (lag experienced)
typed out some text in the same file (lag experienced)
turned off autocomplete
used arrow keys to move cursor up and down within a file (no lag experienced)
typed out some text in the same file (no lag experienced)

hope this helps!

**Comment by arafatkatze** [comment]
@justinechang39 That was incredibly helpful and I could notice the behavior your described pretty clearly on the profiler too. 

Can you confirm two things
- Your Cody Version(You can see it when you go on VsCode Extensions )
<img width="1541" alt="image" src="https://github.com/user-attachments/assets/ff320192-b32c-402a-9703-07365d5dc036">

- Your OS version and name

**Comment by freddyhm** [comment]
I'm also seeing this issue, had to turn autocomplete off.

**Comment by beyang** [comment]
This should be fixed now, please update to 1.36.1 (the latest stable release)!

**Comment by justinechang39** [comment]
[CPU-20241004T082701.787Z.cpuprofile.txt](https://github.com/user-attachments/files/17255488/CPU-20241004T082701.787Z.cpuprofile.txt)

hello! I just tested this again and I'm still experiencing the stuttering.
I have attached a new profile output.

Events:
Before profiling began autocomplete was OFF
Used arrow to navigate cursor up and down in a file (no lag)
wrote some words (no lag)
autocomplete turned ON
Used arrow to navigate cursor up and down in a file (lag)
wrote some words (lag)
autocomplete turned OFF
Used arrow to navigate cursor up and down in a file (no lag)
wrote some words (no lag)

Extension version: v1.37.1727968618 (pre-release)
VSCode version: Version: 1.94.0 (Universal)
MacOS version: Version 15.1 Beta (24B5055e) (Sequoia)
